src/main/java/com/mongodb/hibernate/jdbc/MongoPreparedStatement.java

```diff
@@ -245,35 +227,35 @@ public void addBatch() throws SQLException {
     }
 
     @Override
-    public void setArray(int parameterIndex, @Nullable Array x) throws SQLException {
+    public void setArray(int parameterIndex, Array x) throws SQLException {
         checkClosed();
         checkParameterIndex(parameterIndex);
         throw new NotYetImplementedException();
     }
 
     @Override
-    public void setDate(int parameterIndex, @Nullable Date x, @Nullable Calendar cal) throws SQLException {
+    public void setDate(int parameterIndex, Date x, Calendar cal) throws SQLException {
         checkClosed();
         checkParameterIndex(parameterIndex);
         throw new NotYetImplementedException("To implement in scope of https://jira.mongodb.org/browse/HIBERNATE-42");
     }
 
     @Override
-    public void setTime(int parameterIndex, @Nullable Time x, @Nullable Calendar cal) throws SQLException {
+    public void setTime(int parameterIndex, Time x, Calendar cal) throws SQLException {
         checkClosed();
         checkParameterIndex(parameterIndex);
         throw new NotYetImplementedException("To implement in scope of https://jira.mongodb.org/browse/HIBERNATE-42");
     }
 
     @Override
-    public void setTimestamp(int parameterIndex, @Nullable Timestamp x, @Nullable Calendar cal) throws SQLException {
+    public void setTimestamp(int parameterIndex, Timestamp x, Calendar cal) throws SQLException {
         checkClosed();
         checkParameterIndex(parameterIndex);
         throw new NotYetImplementedException("To implement in scope of https://jira.mongodb.org/browse/HIBERNATE-42");
     }
 
     @Override
-    public void setNull(int parameterIndex, int sqlType, @Nullable String typeName) throws SQLException {
+    public void setNull(int parameterIndex, int sqlType, String typeName) throws SQLException {
         checkClosed();
         checkParameterIndex(parameterIndex);
         throw new NotYetImplementedException("To be implemented during Array / Struct tickets");
```

Comment:
I checked the Javadoc to ensure we should not use `@Nullable`; also, Hibernate would have invoke the `setNull()` method for null value. See https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/type/descriptor/jdbc/BasicBinder.java for further details.